be/src/olap/rowset/segment_v2/page_handle.h

```diff
@@ -46,42 +46,49 @@ class PageHandle {
             : _is_data_owner(false), _cache_data(std::move(cache_data)) {}
 
     // Move constructor
-    PageHandle(PageHandle&& other) noexcept
-            : _is_data_owner(false),
-              _data(std::move(other._data)),
-              _cache_data(std::move(other._cache_data)) {
+    PageHandle(PageHandle&& other) noexcept : _cache_data(std::move(other._cache_data)) {
         // we can use std::exchange if we switch c++14 on
         std::swap(_is_data_owner, other._is_data_owner);
+        _data.swap(other._data);
     }
 
     PageHandle& operator=(PageHandle&& other) noexcept {
         std::swap(_is_data_owner, other._is_data_owner);
-        _data = std::move(other._data);
+        _data.swap(other._data);
         _cache_data = std::move(other._cache_data);
         return *this;
     }
 
     ~PageHandle() {
         if (_is_data_owner && _data.size > 0) {
-            delete[] _data.data;
             ExecEnv::GetInstance()->page_no_cache_mem_tracker()->consume(-_data.size);
```

Comment:
if this page handle is data owner, why not call delete to release the memory?

Reply:
Because DataPage will automatically call `free` when it is destructed

The memory life cycle is the same as DataPage, and does not rely on manual release, which seems to be more exception-safe